storage/database/dynamodb_test.go

```diff
@@ -242,7 +242,9 @@ func testDynamoBatch_WriteMutliTables(t *testing.T) {
 }
 
 func (dynamo *dynamoDB) deleteDB() {
-	dynamo.Close()
+	if !dynamo.config.ReadOnly {
+		dynamo.Close()
```

Comment:
Why do we close this in read only case?
How about checking this in `Close()`?

Reply:
review reflected